tests/regressions/lcos/shared_future_then_with_executor_3634.cpp

```diff
@@ -0,0 +1,25 @@
+//  Copyright (c) 2019 Christopher Hinz
+//
+//  Distributed under the Boost Software License, Version 1.0. (See accompanying
+//  file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+// This test case demonstrates the issue described in #3634:
+// The build fails if shared_future<>::then is called with a thread executor
+
+#include <hpx/hpx_main.hpp>
+
+#include <hpx/lcos/future.hpp>
+
+#include <hpx/parallel/executors.hpp> // Workaround for a missing header file
+#include <hpx/runtime/threads/executors/pool_executor.hpp>
+
+int main()
+{
+    hpx::threads::executors::pool_executor executor("default");
+
+    auto future = hpx::make_ready_future().share();
+
+    future.then(executor, [](auto future) { future.get(); });
```

Comment:
Could you change this to `hpx::shared_future<void>` instead of `auto`. `auto` is not allowed with C++11 (http://cdash.cscs.ch//viewBuildError.php?buildid=37278).

Reply:
Of course I can. Thanks for the remark.